### PR TITLE
Fix failing xgboost test in the cudf.pandas third-party integration tests

### PR DESCRIPTION
**Pull request by Matt711**
## Description
A part of #17490
## Checklist
- [ ] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

File: .github/workflows/pr.yaml
Comment (bdice) [inline .github/workflows/pr.yaml:334]:
```suggestion
    uses: rapidsai/shared-workflows/.github/workflows/custom-job.yaml@branch-25.02
```

File: python/cudf/cudf_pandas_tests/third_party_integration_tests/tests/test_xgboost.py
Comment (bdice) [inline python/cudf/cudf_pandas_tests/third_party_integration_tests/tests/test_xgboost.py:130]: Is this TODO something we need to change in cudf.pandas, or is it a bug in xgboost?

Comment (Matt711) [reply python/cudf/cudf_pandas_tests/third_party_integration_tests/tests/test_xgboost.py:130]: We'd need to make a change in xgboost. Let me cc @galipremsagar to get his thoughts. I believe `.inplace_predict()` should return a cudf.pandas proxy ndarray.

Comment (Matt711) [reply python/cudf/cudf_pandas_tests/third_party_integration_tests/tests/test_xgboost.py:130]: Related issue: https://github.com/rapidsai/cudf/issues/17524

File: .github/workflows/pr.yaml
Comment (Matt711) [inline .github/workflows/pr.yaml:42]:
```suggestion
      # TODO: Remove this CI job after https://github.com/rapidsai/cudf/issues/17490 is resolved
      # - third-party-integration-tests-cudf-pandas
```

File: python/cudf/cudf_pandas_tests/third_party_integration_tests/tests/test_xgboost.py
Comment (galipremsagar) [reply python/cudf/cudf_pandas_tests/third_party_integration_tests/tests/test_xgboost.py:130]: I don't think we can make `.inplace_predict()` return a `cudf.pandas` proxy object. Xgboost takes different code-paths for `cudf` & `pandas` objects, for a library that executes object(gpu/cpu) aware, I don't think there is much we can do rather than the current state of fixes present in xgboost. But I also feel `predict` has very old code and it needs to be updated to match `inplace_predict` behavior. So I opened a pr asking for reviews: https://github.com/dmlc/xgboost/pull/11118

File: python/cudf/cudf_pandas_tests/third_party_integration_tests/tests/test_xgboost.py
Comment (Matt711) [inline python/cudf/cudf_pandas_tests/third_party_integration_tests/tests/test_xgboost.py:131]:
```suggestion
```

**Review by bdice — APPROVED** [review]
Approving, assuming we revert the workflow changes before merge.

File: .github/workflows/pr.yaml
Comment (Matt711) [inline .github/workflows/pr.yaml:42]:
```suggestion
```

File: .github/workflows/pr.yaml
Comment (Matt711) [inline .github/workflows/pr.yaml:341]:
```suggestion
```

**Pull request merged**
